ee/modules/160-multitenancy-manager/hooks/project.go

```diff
@@ -0,0 +1,192 @@
+
+		ptSpecValues, ok := input.Values.GetOk(internal.ModuleValuePath(internal.PTValuesPath, project.ProjectTypeName))
+		if !ok {
+			errMsg := fmt.Sprintf("Can't find valid ProjectType '%s' for Project", project.ProjectTypeName)
+			errStatusProject(errMsg)
+			continue
+		}
+
+		ptValues := ptSpecValues.Value()
+		ptValuesMap, ok := ptValues.(map[string]interface{})
+		if !ok {
+			errMsg := fmt.Sprintf("can't convert '%s' ProjectType values to map[string]interface: %T", project.ProjectTypeName, ptValues)
+			errStatusProject(errMsg)
+			continue
+		}
+
+		schema, err := internal.LoadOpenAPISchema(ptValuesMap["openAPI"])
+		if err != nil {
+			errMsg := fmt.Sprintf("can't load '%s' ProjectType OpenAPI schema: %v", project.ProjectTypeName, err)
+			errStatusProject(errMsg)
+			continue
+		}
+
+		projectTemplateValues, err := mapStringJSONToMapStringInterface(project.Template)
+		if err != nil {
+			errMsg := fmt.Sprintf("can't convert map[string]*apiext.JSON to map[string]interface{} for template values: %v", err)
+			errStatusProject(errMsg)
+			continue
+		}
+
+		if schema.AdditionalProperties == nil {
+			schema.AdditionalProperties = &spec.SchemaOrBool{Allows: false}
+		}
+		if err := validate.AgainstSchema(schema, projectTemplateValues, strfmt.Default); err != nil {
+			errMsg := fmt.Sprintf("template data didn't match OpenAPI schema for '%s' ProjectType: %v", project.ProjectTypeName, err)
+			errStatusProject(errMsg)
+			continue
+		}
+
+		values = append(values, projectValues{
+			ProjectTypeName: project.ProjectTypeName,
+			ProjectName:     project.Name,
+			Params:          projectTemplateValues,
+		})
+
+		setSyncStatusProject(input.PatchCollector, project.Name, project.Conditions)
+	}
+
+	input.Values.Set(internal.ModuleValuePath(internal.ProjectValuesPath), values)
+	return nil
+}
+
+func setErrorProjectWrap(patcher *object_patch.PatchCollector, projectName string, conditions []v1alpha1.Condition) func(errMsg string) {
+	return func(errMsg string) {
+		setErrorStatusProject(patcher, projectName, errMsg, conditions)
+	}
+}
+
+func setErrorStatusProject(patcher *object_patch.PatchCollector, projectName, errMsg string, conditions []v1alpha1.Condition) {
+	conditions = append(conditions, v1alpha1.Condition{
+		Name:    "Error",
+		Message: errMsg,
+		Status:  false,
+	})
+
+	internal.SetProjectStatus(patcher, projectName, false, errMsg, conditions)
+}
+
+func setSyncStatusProject(patcher *object_patch.PatchCollector, projectName string, conditions []v1alpha1.Condition) {
+	conditions = append(conditions, v1alpha1.Condition{
+		Name:   "Sync",
+		Status: true,
+	})
+	internal.SetProjectStatus(patcher, projectName, true, "", conditions)
+}
+
+func mapStringJSONToMapStringInterface(m map[string]*apiext.JSON) (map[string]interface{}, error) {
+	mm, err := json.Marshal(m)
+	if err != nil {
+		return nil, err
+	}
+
+	var resultMap map[string]interface{}
```

Comment:
```suggestion
	var result map[string]interface{}
```
NIT: Usually it is clear from the type of the variable. This is better to avoid `var connectionString string = "https://..."`